train.py

```diff
@@ -42,30 +59,46 @@ def train(
     optimizer = optim.Adam(model.parameters())
 
     iteration_times = []
-    for iteration, batch in enumerate(dl):
-        texts, texts_mask, strokes, strokes_mask = batch
-        if iteration >= num_training_iterations:
-            break
-        optimizer.zero_grad()
-        start = time.time()
-        strokes = strokes.to(device)
-        output = model(strokes)
-        iteration_times.append(time.time() - start)
-
-        # output[0] is the prediction, strokes[1] is the ground truth
-        loss = ((strokes[1:] - output[:-1]) ** 2).mean()
-        if (iteration % 10) == 0:
-            print(f"iter {iteration:4d}: {loss.item()=:.4f}")
-        writer.add_scalar('Loss/train', loss.item(), iteration)
-        loss.backward()
-        optimizer.step()
+    with tqdm(total=num_training_iterations) as pbar:
+        for iteration, batch in enumerate(dl):
+            texts, texts_mask, strokes, strokes_mask = batch
+            if iteration >= num_training_iterations:
+                break
+            optimizer.zero_grad()
+            start = time.time()
+            strokes = strokes.to(device)
```

Comment:
Could we add this to the dataloader? Basically that all the `to` calls happen in the dataloader?

Reply:
Wouldn't we need to still do `model.to()`?

Reply:
yes for `model.to` but all the data can be done in the dataloader. Here's an example:
https://stackoverflow.com/questions/65932328/pytorch-while-loading-batched-data-using-dataloader-how-to-transfer-the-data-t

Reply:
Done